### PR TITLE
Fix: correctly filter zone metadata

### PR DESCRIPTION
**Pull request by caspiano**
Correctly filter metadata by name.
Previously it would the parent's metadata for the children query.
Also added a few tests.

### CODE REVIEW

File: shard.lock [hunk 1]
```diff
@@ -53,7 +53,7 @@ shards:
     version: 0.1.4
 
   etcd:
-    git: https://github.com/place-labs/crystal-etcd.git
+    git: https://github.com/aca-labs/crystal-etcd.git
```
Comment (stakach) [inline shard.lock:56]: was the revert to aca-labs intentional?

File: shard.lock [hunk 2]
```diff
@@ -201,7 +201,7 @@ shards:
     version: 1.1.0
 
   ulid:
-    git: https://github.com/place-labs/ulid.git
+    git: https://github.com/aca-labs/ulid.git
```
Comment (stakach) [inline shard.lock:204]: Same here

File: shard.lock (hunk 1 above)
Comment (caspiano) [reply shard.lock:56]: nope, i don't know where they're coming potentially some weird name resolution?

Comment (caspiano) [reply shard.lock:56]: either way, same version. i think they'll probably revert if the versions are pinned

Comment (stakach) [reply shard.lock:56]: ahh yeah it redirects, all good

**Pull request merged**